2 dz/6/main.cpp

```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+#include <cstdlib>
+#include <ctime>
+#include <cmath>
+
+using namespace std;
+
+void num_gen(int *hidden_number, int size)
+{
+    srand(time(NULL));
+    for (int i = 0; i < size; i++)
+    {
+        hidden_number[i] = rand() % 10;
+        if (hidden_number[0] == 0)
+        {
+            i = -1;
+            break;
+        }
+        if (i > 0)
+        {
+            for (int j = i - 1; j >= 0; j--)
+            {
+                if (hidden_number[i] == hidden_number[j])
```

Comment:
Do I understand correctly that if someone wants to generate a number with a length of `size`, they can get a surprise?
And what if `size` > 10? Should we do something arbitrary or throw the exception? I would prefer the second.